File: java/src/modal/Reservation.java
package modal;

import java.time.LocalDate;

public class Reservation {
    public final LocalDate startDate;
    public final LocalDate endDate;

    public Reservation(LocalDate startDate, LocalDate endDate) {
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public boolean isReservationValid(Reservation other) {
        return (this.endDate.isBefore(other.startDate) || this.startDate.isAfter(other.endDate));
    }


}

File: python/modal/reservation.py
class Reservation:
    def __init__(self, start_date, end_date):
        self.start_date = start_date
        self.end_date = end_date

    def is_valid(self, other):
        return self.end_date < other.start_date or self.start_date > other.end_date
